packages/data-broker/src/tests/services/weather/WeatherService.test.js

```diff
@@ -0,0 +1,234 @@
+import {
+  createMockModelsStubWithMockEntity,
+  createWeatherApiStub,
+  createWeatherApiStubWithMockResponse,
+  getMockDataSourcesArg,
+  getMockOptionsArg,
+  getMockTypeArg,
+} from './WeatherService.stubs';
+import { mockNow, resetMocks } from './testutil';
+
+describe('WeatherService', () => {
+  afterEach(() => {
+    resetMocks();
+  });
+
+  describe('basic operation', () => {
+    it('returns analytics data when requesting data for dataElements', async () => {
+      const mockModels = await createMockModelsStubWithMockEntity();
+
+      const mockApiResponse = {
+        data: [
+          {
+            precip: 23.6,
+            max_temp: 29.8,
+            min_temp: 24,
+            datetime: '2020-08-20',
+          },
+          {
+            precip: 5,
+            max_temp: 6,
+            min_temp: 7,
+            datetime: '2020-08-21',
+          },
+        ],
+        sources: [],
+      };
+
+      const mockApi = createWeatherApiStub(mockApiResponse);
+
+      const service = new WeatherService(mockModels, mockApi);
+
+      const actual = await service.pull(
+        [
+          {
+            model: {},
+            id: '12345_PRECIP',
+            code: 'PRECIP',
+            type: 'dataElement',
+            service_type: 'weather',
+            config: {},
+          },
+        ],
+        'dataElement',
+        getMockOptionsArg(),
+      );
+
+      expect(actual.results).to.deep.equal([
+        { dataElement: 'PRECIP', value: 23.6, organisationUnit: 'MELB', period: '20200820' },
+        { dataElement: 'PRECIP', value: 5, organisationUnit: 'MELB', period: '20200821' },
+      ]);
+    });
+
+    it('returns events data when requesting data for a dataGroup', async () => {
+      const mockModels = await createMockModelsStubWithMockEntity();
+
+      const mockApiResponse = {
+        data: [
+          {
+            precip: 23.6,
+            max_temp: 29.8,
+            min_temp: 24,
+            datetime: '2020-08-20',
+          },
+          {
+            precip: 5,
+            max_temp: 6,
+            min_temp: 7,
+            datetime: '2020-08-21',
+          },
+        ],
+        sources: [],
+      };
+
+      const mockApi = createWeatherApiStub(mockApiResponse);
+
+      const service = new WeatherService(mockModels, mockApi);
+
+      const actual = await service.pull(
+        [
+          {
+            model: {},
+            id: 'SOME_DATA_GROUP_ID', // data group doesn't matter, it will return the data elements it knows about
+            code: 'SOME_DATA_GROUP_CODE',
+            type: 'dataGroup',
+            service_type: 'weather',
+            config: {},
+          },
+        ],
+        'dataGroup',
+        getMockOptionsArg(),
+      );
+
+      expect(actual.results).to.deep.equal([
+        {
+          organisationUnit: 'MELB',
+          PRECIP: 23.6,
+          MAX_TEMP: 29.8,
+          MIN_TEMP: 24,
+          period: '20200820',
```

Comment:
I think the format of events is more like
```suggestion
          event: 'someId',
          orgUnit: 'MELB',
          orgUnitName: 'Melbourne',
          dataValues: {
            PRECIP: 23.6,
            MAX_TEMP: 29.8,
            MIN_TEMP: 24,
          },
          eventDate: '2020-08-20T23:59:59',
```

Reply:
Sounds good. Q) Do you think `eventDate: '2020-08-20T23:59:59',` makes sense having the time component? The most specific time we can get for the event is at the day level, max temp / min temp are reached at different times, so it might be confusing having a time component in there at 23:59:59. Thoughts?

Reply:
That's a good point about them being reached at different times during the day. The very end of the day, i.e. 23:59:59 , still makes sense to me though - it's at that point that we know we can lock in the max and min. And I do think we should have a time component for consistency, in case a downstream function expects it. Valid questions to raise though, and @kael89 (who has put a lot of thought into the difference between event and analytic formats) may have additional comments

Reply:
I agree with Edwin's points:

✔️ we should have a time for consistency. Possibly we should also include a timezone - I'm gonna check this now
✔️ we should use the most accurate time to our knowledge. At this use case, the last minute of the day makes sense, since the event reflects a summary of the day

We haven't been using time till now, but you never know in the future. It's also to be consistent for predictable field parsing etc

Reply:
I confirmed that no timezone info is included in event dates

Reply:
Updated PR with 23:59:59